File: VB/09_Forms/ModifyFormFieldValue.vb
Imports Spire.Pdf
Imports Spire.Pdf.Widget
Imports Spire.Pdf.Fields

Namespace ModifyFormFieldValue
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Specify the input file path
			Dim input As String = "..\..\..\..\..\..\Data\FormField.pdf"

			' Create a new instance of PdfDocument
			Dim pdf As New PdfDocument()
			pdf.LoadFromFile(input)

			' Access the form widget of the PDF document
			Dim form As PdfFormWidget = TryCast(pdf.Form, PdfFormWidget)

			' Iterate through each field in the form
			For i As Integer = 0 To form.FieldsWidget.List.Count - 1
				Dim field As PdfField = TryCast(form.FieldsWidget.List(i), PdfField)

				' Check if the field is a text box
				If TypeOf field Is PdfTextBoxFieldWidget Then
					Dim textbox As PdfTextBoxFieldWidget = TryCast(field, PdfTextBoxFieldWidget)

					' Check if the text box has a specific name (e.g., "TextBox1")
					If textbox.Name = "TextBox1" Then
						' Modify the text value of the text box
						textbox.Text = "New value"
					End If
				End If
			Next i

			' Specify the output file path
			Dim result As String = "ModifyFormFieldValue_out.pdf"

			' Save the modified PDF document to the output file
			pdf.SaveToFile(result)

			' Close the document
			pdf.Close()

			' Launch the Pdf file
			PDFDocumentViewer(result)
		End Sub
		Private Sub PDFDocumentViewer(ByVal filename As String)
			Try
				Process.Start(filename)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS/09_Forms/ModifyFormFieldValue.cs
using System;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Widget;
using Spire.Pdf.Fields;
namespace ModifyFormFieldValue
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Specify the path to the input PDF file.
            string input = @"..\..\..\..\..\..\Data\FormField.pdf";

            // Open the PDF document.
            PdfDocument pdf = new PdfDocument();
            pdf.LoadFromFile(input);

            // Get the form widget from the PDF document.
            PdfFormWidget form = pdf.Form as PdfFormWidget;

            // Iterate through each field in the form.
            for (int i = 0; i < form.FieldsWidget.List.Count; i++)
            {
                // Get the current field.
                PdfField field = form.FieldsWidget.List[i] as PdfField;

                // Check if the field is a TextBox field.
                if (field is PdfTextBoxFieldWidget)
                {
                    // Convert the field to a TextBox field.
                    PdfTextBoxFieldWidget textbox = field as PdfTextBoxFieldWidget;

                    // Check if the TextBox field has a specific name.
                    if (textbox.Name == "TextBox1")
                    {
                        // Modify the text value of TextBox1.
                        textbox.Text = "New value";
                    }
                }
            }

            // Specify the path for the output PDF file.
            String result = "ModifyFormFieldValue_out.pdf";

            // Save the modified PDF document to the output file.
            pdf.SaveToFile(result);

            //Launch the Pdf file
            PDFDocumentViewer(result);
        }
        private void PDFDocumentViewer(string filename)
        {
            try
            {
                System.Diagnostics.Process.Start(filename);
            }
            catch { }
        }
    }
}
